### PR TITLE
Add image prune

### LINKED ISSUE #64: Delete all unused images
Is it possible to delete all images that are not currently being used?

### PR DESCRIPTION
**Pull request by cpuguy83**
**What this PR does / why we need it**:

A few cleanup things, but mainly it adds support for pruning all non-running images.
This is done by specifying `*` in an imagelist.

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by sozercan** [comment]
can we add a test for this

**Comment by cpuguy83** [comment]
Added e2e test.
I'd recommend reviewing each commit individually to understand the context for the change.

File: test/e2e/eraser_test.go [hunk 1]
```diff
@@ -187,8 +126,168 @@ func TestRemoveImagesFromAllNodes(t *testing.T) {
...
+			if err = wait.For(conditions.New(client.Resources()).DeploymentConditionMatch(&caddyDep, appsv1.DeploymentAvailable, corev1.ConditionTrue),
+				wait.WithTimeout(time.Minute*1)); err != nil {
+				t.Fatal("caddy deployment not found", err)
+			}
+			ctx = context.WithValue(ctx, caddy, &caddyDep)
+
+			return ctx
+		}).
+		Assess("Remove some of the deployments so the images aren't running", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			// Here we remove the rdis and caddy deployments
```
Comment (sozercan) [inline test/e2e/eraser_test.go]: nit: typo

File: test/e2e/eraser_test.go [hunk 2]
```diff
@@ -187,8 +126,168 @@ func TestRemoveImagesFromAllNodes(t *testing.T) {
...
+			// The first check could take some extra time, where as things should be done already for the 2nd check.
+			// So we'll give plenty of time and fail slow here.
+			ctxT, cancel := context.WithTimeout(ctx, 5*time.Minute)
+			defer cancel()
+			checkImageRemoved(ctxT, t, getClusterNodes(t), redis)
+
+			ctxT, cancel = context.WithTimeout(ctx, time.Minute)
+			defer cancel()
+			checkImageRemoved(ctxT, t, getClusterNodes(t), caddy)
+
```
Comment (sozercan) [inline test/e2e/eraser_test.go:260]: do we want to check if nginx is still there?

Comment (cpuguy83) [reply test/e2e/eraser_test.go:260]: Done.

**Review by sozercan — APPROVED** [review]
LGTM

**Pull request merged**